### PR TITLE
test(locale-modal): add additional e2e tests

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
### Related Ticket(s)
#7432 

### Description
Added additional tests for the local modal component.

### Changelog

**New**

- new tests for the locale-modal tests

### CODE REVIEW

File: packages/react/tests/e2e-storybook/cypress/integration/LocaleModal/LocaleModal.e2e.js [hunk 1]
```diff
@@ -20,27 +20,75 @@ describe('LocaleModal | default', () => {
...
+
   it('should load the Americas region', () => {
     cy.get('[data-region="am"]').click();
+    cy.get('.bx--locale-modal__locales').should('have.length', 35);
 
     cy.screenshot();
 
     // Take a snapshot for visual diffing
-    cy.percySnapshot('LocaleModal | region selected', {
+    cy.percySnapshot('Footer | Region selected', {
```
Comment (annawen1) [inline packages/react/tests/e2e-storybook/cypress/integration/LocaleModal/LocaleModal.e2e.js:45]: should this be part of Footer e2e tests then? or should we change the snapshot name back to "LocaleModal"

File: packages/react/tests/e2e-storybook/cypress/integration/LocaleModal/LocaleModal.e2e.js [hunk 2]
```diff
@@ -20,27 +20,75 @@ describe('LocaleModal | default', () => {
...
+    )
+      .first()
+      .then(e => {
+        expect(e.text()).to.equal('Mexico');
+      });
+
+    cy.screenshot();
+
+    // Take a snapshot for visual diffing
+    cy.percySnapshot('Footer | Filtering locales', {
```
Comment (annawen1) [inline packages/react/tests/e2e-storybook/cypress/integration/LocaleModal/LocaleModal.e2e.js:67]: should we also change the snapshot name to "LocaleModal" here?

**Review by annawen1 — COMMENTED** [review]
Looks good, just some snapshot name change to LocaleModal

File: packages/react/tests/e2e-storybook/cypress/integration/LocaleModal/LocaleModal.e2e.js (hunk 1 above)
Comment (IgnacioBecerra) [reply packages/react/tests/e2e-storybook/cypress/integration/LocaleModal/LocaleModal.e2e.js:45]: Yeah the footer is also testing some of the LocaleModal functions, but I changed to what it should be here 😅

**Review by annawen1 — APPROVED** [review]
LGTM!

**Pull request merged**